boards/components/src/process_info_driver.rs

```diff
@@ -0,0 +1,58 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2025.
+
+//! Component for the process info capsule.
+
+use capsules_extra::process_info_driver::{self, ProcessInfo};
+use core::mem::MaybeUninit;
+use kernel::capabilities;
+use kernel::component::Component;
+use kernel::create_capability;
+
+#[macro_export]
+macro_rules! process_info_component_static {
+    () => {{
+        let process_info = kernel::static_buf!(
+            capsules_extra::process_info_driver::ProcessInfo<
+                components::process_info_driver::Capability,
+            >
+        );
+
+        process_info
+    };};
+}
+
+pub struct ProcessInfoComponent {
+    board_kernel: &'static kernel::Kernel,
+    driver_num: usize,
+}
+
+impl ProcessInfoComponent {
+    pub fn new(board_kernel: &'static kernel::Kernel, driver_num: usize) -> Self {
+        Self {
+            board_kernel,
+            driver_num,
+        }
+    }
+}
+
+pub struct Capability;
```

Comment:
I noticed something while looking at the Rust compiler recently. They include a private field into structs such that they cannot be (safely) constructed from outside the current crate:

https://doc.rust-lang.org/beta/nightly-rustc/rustc_span/struct.ErrorGuaranteed.html

 This seems desirable and important for the safety of Tock's capabilities as well. If they're public types, then anyone can name them, and we must prevent them from being constructed without the use of an unsafe constructor.

I'll check our other uses of capabilities whether that's a problem we have across the Tock kernel.